sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -950,6 +950,10 @@ def run_ParDo(self, transform_node, options):
       step.add_property(PropertyNames.RESTRICTION_ENCODING,
                         self._get_cloud_encoding(restriction_coder))
 
+    from apache_beam.runners.common import DoFnSignature
+    if DoFnSignature(transform.dofn).is_stateful_dofn():
```

Comment:
This will be only needed for streaming pipelines.

Reply:
should we distinguish streaming pipelines from batch here?

Reply:
Yes, java side we do the same.

https://github.com/apache/beam/blob/2fd785dedb979a248e63c6385f978fd18fd2fbc4/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java#L1260

Reply:
I see, done.